### PR TITLE
[azure] set DES in defaultMachinePlatform 

### PR DESCRIPTION
**Pull request by jinyunma**
[CORS-2080](https://issues.redhat.com/browse/CORS-2080) Extend user encryption keys usage for the default storage

### CODE REVIEW

**Comment by jinyunma** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installer-rehearse-4.13-installer-rehearse-azure

**Comment by jinyunma** [comment]
cc @jianlinliu @gpei @yunjiang29 to review

File: ci-operator/step-registry/ipi/conf/azure/provisioned-des/ipi-conf-azure-provisioned-des-commands.sh [hunk 1]
```diff
@@ -27,22 +27,13 @@ PATCH="/tmp/install-config-provisioned-des.yaml.patch"
 
 # create a patch with existing des
 cat > "${PATCH}" << EOF
-compute:
-- platform:
-    azure:
+platform:
+  azure:
+    defaultMachinePlatform:
```
Comment (jianlinliu) [inline ci-operator/step-registry/ipi/conf/azure/provisioned-des/ipi-conf-azure-provisioned-des-commands.sh:32]: This change can be applied to older version of OCP?

Comment (yunjiang29) [reply ci-operator/step-registry/ipi/conf/azure/provisioned-des/ipi-conf-azure-provisioned-des-commands.sh:32]: Jinyun, I think you can consider introducing a new var to control whether use `compute/controlPlane` or `defaultMachinePlatform`, the default would be the _existing behavior_ .

**Comment by jinyunma** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.11-amd64-nightly-azure-ipi-disk-encryption-set-p2-f14

**Comment by jinyunma** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installer-rehearse-4.13-installer-rehearse-azure

**Comment by jinyunma** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.11-amd64-nightly-azure-ipi-disk-encryption-set-p2-f14

**Comment by jinyunma** [comment]
Checked two jobs:
on 4.13, diskEncryptionSet setting is under defaultMachinePlatform:
```
platform:
  azure:
    region: southcentralus
    baseDomainResourceGroupName: os4-common
    defaultMachinePlatform:
      encryptionAtHost: true
      osDisk:
        diskEncryptionSet:
          resourceGroup: ci-op-dbhs2fdx-064eb-rg
          name: ci-op-dbhs2fdx-064eb-des
controlPlane:
  architecture: amd64
  name: master
  platform:
    azure:
      type: null
compute:
- architecture: amd64
  name: worker
  replicas: 3
  platform:
    azure:
      type: Standard_D4s_v3


$ az vm list -g ci-op-dbhs2fdx-064eb-wn875-rg --query "[].[id,storageProfile.osDisk.managedDisk.diskEncryptionSet,securityProfile.encryptionAtHost]"
[
  [
    "/subscriptions/53b8f551-f0fc-4bea-8cba-6d1fefd54c8a/resourceGroups/ci-op-dbhs2fdx-064eb-wn875-rg/providers/Microsoft.Compute/virtualMachines/ci-op-dbhs2fdx-064eb-wn875-master-0",
    {
      "id": "/subscriptions/53b8f551-f0fc-4bea-8cba-6d1fefd54c8a/resourceGroups/ci-op-dbhs2fdx-064eb-rg/providers/Microsoft.Compute/diskEncryptionSets/ci-op-dbhs2fdx-064eb-des",
      "resourceGroup": "ci-op-dbhs2fdx-064eb-rg"
    },
    true
  ],
  [
    "/subscriptions/53b8f551-f0fc-4bea-8cba-6d1fefd54c8a/resourceGroups/ci-op-dbhs2fdx-064eb-wn875-rg/providers/Microsoft.Compute/virtualMachines/ci-op-dbhs2fdx-064eb-wn875-master-1",
    {
      "id": "/subscriptions/53b8f551-f0fc-4bea-8cba-6d1fefd54c8a/resourceGroups/ci-op-dbhs2fdx-064eb-rg/providers/Microsoft.Compute/diskEncryptionSets/ci-op-dbhs2fdx-064eb-des",
      "resourceGroup": "ci-op-dbhs2fdx-064eb-rg"
    },
    true
  ],
  [
    "/subscriptions/53b8f551-f0fc-4bea-8cba-6d1fefd54c8a/resourceGroups/ci-op-dbhs2fdx-064eb-wn875-rg/providers/Microsoft.Compute/virtualMachines/ci-op-dbhs2fdx-064eb-wn875-master-2",
    {
      "id": "/subscriptions/53b8f551-f0fc-4bea-8cba-6d1fefd54c8a/resourceGroups/ci-op-dbhs2fdx-064eb-rg/providers/Microsoft.Compute/diskEncryptionSets/ci-op-dbhs2fdx-064eb-des",
      "resourceGroup": "ci-op-dbhs2fdx-064eb-rg"
    },
    true
  ],
  [
    "/subscriptions/53b8f551-f0fc-4bea-8cba-6d1fefd54c8a/resourceGroups/ci-op-dbhs2fdx-064eb-wn875-rg/providers/Microsoft.Compute/virtualMachines/ci-op-dbhs2fdx-064eb-wn875-worker-southcentralus1-q9b7c",
    {
      "id": "/subscriptions/53b8f551-f0fc-4bea-8cba-6d1fefd54c8a/resourceGroups/ci-op-dbhs2fdx-064eb-rg/providers/Microsoft.Compute/diskEncryptionSets/ci-op-dbhs2fdx-064eb-des",
      "resourceGroup": "ci-op-dbhs2fdx-064eb-rg"
    },
    true
  ],
  [
    "/subscriptions/53b8f551-f0fc-4bea-8cba-6d1fefd54c8a/resourceGroups/ci-op-dbhs2fdx-064eb-wn875-rg/providers/Microsoft.Compute/virtualMachines/ci-op-dbhs2fdx-064eb-wn875-worker-southcentralus2-8r657",
    {
      "id": "/subscriptions/53b8f551-f0fc-4bea-8cba-6d1fefd54c8a/resourceGroups/ci-op-dbhs2fdx-064eb-rg/providers/Microsoft.Compute/diskEncryptionSets/ci-op-dbhs2fdx-064eb-des",
      "resourceGroup": "ci-op-dbhs2fdx-064eb-rg"
    },
    true
  ],
  [
    "/subscriptions/53b8f551-f0fc-4bea-8cba-6d1fefd54c8a/resourceGroups/ci-op-dbhs2fdx-064eb-wn875-rg/providers/Microsoft.Compute/virtualMachines/ci-op-dbhs2fdx-064eb-wn875-worker-southcentralus3-scntc",
    {
      "id": "/subscriptions/53b8f551-f0fc-4bea-8cba-6d1fefd54c8a/resourceGroups/ci-op-dbhs2fdx-064eb-rg/providers/Microsoft.Compute/diskEncryptionSets/ci-op-dbhs2fdx-064eb-des",
      "resourceGroup": "ci-op-dbhs2fdx-064eb-rg"
    },
    true
  ]
]
```
On 4.11, diskEncryptionSet is under contraolPlane/compute:
```
platform:
  azure:
    region: southcentralus
    baseDomainResourceGroupName: os4-common
controlPlane:
  architecture: amd64
  name: master
  platform:
    azure:
      type: null
      encryptionAtHost: true
      osDisk:
        diskEncryptionSet:
          resourceGroup: ci-op-712c71yw-ac3fd-rg
          name: ci-op-712c71yw-ac3fd-des
compute:
- architecture: amd64
  name: worker
  replicas: 3
  platform:
    azure:
      type: Standard_D4s_v3
      encryptionAtHost: true
      osDisk:
        diskEncryptionSet:
          resourceGroup: ci-op-712c71yw-ac3fd-rg
          name: ci-op-712c71yw-ac3fd-des


$ az vm list -g ci-op-712c71yw-ac3fd-vthfh-rg --query "[].[id,storageProfile.osDisk.managedDisk.diskEncryptionSet,securityProfile.encryptionAtHost]"
[
  [
    "/subscriptions/53b8f551-f0fc-4bea-8cba-6d1fefd54c8a/resourceGroups/ci-op-712c71yw-ac3fd-vthfh-rg/providers/Microsoft.Compute/virtualMachines/ci-op-712c71yw-ac3fd-vthfh-master-0",
    {
      "id": "/subscriptions/53b8f551-f0fc-4bea-8cba-6d1fefd54c8a/resourceGroups/ci-op-712c71yw-ac3fd-rg/providers/Microsoft.Compute/diskEncryptionSets/ci-op-712c71yw-ac3fd-des",
      "resourceGroup": "ci-op-712c71yw-ac3fd-rg"
    },
    true
  ],
  [
    "/subscriptions/53b8f551-f0fc-4bea-8cba-6d1fefd54c8a/resourceGroups/ci-op-712c71yw-ac3fd-vthfh-rg/providers/Microsoft.Compute/virtualMachines/ci-op-712c71yw-ac3fd-vthfh-master-1",
    {
      "id": "/subscriptions/53b8f551-f0fc-4bea-8cba-6d1fefd54c8a/resourceGroups/ci-op-712c71yw-ac3fd-rg/providers/Microsoft.Compute/diskEncryptionSets/ci-op-712c71yw-ac3fd-des",
      "resourceGroup": "ci-op-712c71yw-ac3fd-rg"
    },
    true
  ],
  [
    "/subscriptions/53b8f551-f0fc-4bea-8cba-6d1fefd54c8a/resourceGroups/ci-op-712c71yw-ac3fd-vthfh-rg/providers/Microsoft.Compute/virtualMachines/ci-op-712c71yw-ac3fd-vthfh-master-2",
    {
      "id": "/subscriptions/53b8f551-f0fc-4bea-8cba-6d1fefd54c8a/resourceGroups/ci-op-712c71yw-ac3fd-rg/providers/Microsoft.Compute/diskEncryptionSets/ci-op-712c71yw-ac3fd-des",
      "resourceGroup": "ci-op-712c71yw-ac3fd-rg"
    },
    true
  ],
  [
    "/subscriptions/53b8f551-f0fc-4bea-8cba-6d1fefd54c8a/resourceGroups/ci-op-712c71yw-ac3fd-vthfh-rg/providers/Microsoft.Compute/virtualMachines/ci-op-712c71yw-ac3fd-vthfh-worker-southcentralus1-j5dcw",
    {
      "id": "/subscriptions/53b8f551-f0fc-4bea-8cba-6d1fefd54c8a/resourceGroups/ci-op-712c71yw-ac3fd-rg/providers/Microsoft.Compute/diskEncryptionSets/ci-op-712c71yw-ac3fd-des",
      "resourceGroup": "ci-op-712c71yw-ac3fd-rg"
    },
    true
  ],
  [
    "/subscriptions/53b8f551-f0fc-4bea-8cba-6d1fefd54c8a/resourceGroups/ci-op-712c71yw-ac3fd-vthfh-rg/providers/Microsoft.Compute/virtualMachines/ci-op-712c71yw-ac3fd-vthfh-worker-southcentralus2-4wbd4",
    {
      "id": "/subscriptions/53b8f551-f0fc-4bea-8cba-6d1fefd54c8a/resourceGroups/ci-op-712c71yw-ac3fd-rg/providers/Microsoft.Compute/diskEncryptionSets/ci-op-712c71yw-ac3fd-des",
      "resourceGroup": "ci-op-712c71yw-ac3fd-rg"
    },
    true
  ],
  [
    "/subscriptions/53b8f551-f0fc-4bea-8cba-6d1fefd54c8a/resourceGroups/ci-op-712c71yw-ac3fd-vthfh-rg/providers/Microsoft.Compute/virtualMachines/ci-op-712c71yw-ac3fd-vthfh-worker-southcentralus3-khqtg",
    {
      "id": "/subscriptions/53b8f551-f0fc-4bea-8cba-6d1fefd54c8a/resourceGroups/ci-op-712c71yw-ac3fd-rg/providers/Microsoft.Compute/diskEncryptionSets/ci-op-712c71yw-ac3fd-des",
      "resourceGroup": "ci-op-712c71yw-ac3fd-rg"
    },
    true
  ]
]
```

**Comment by jinyunma** [comment]
@jianlinliu @yunjiang29 @gpei please help to review again, thanks.

File: ci-operator/step-registry/ipi/conf/azure/provisioned-des/ipi-conf-azure-provisioned-des-ref.yaml [hunk 2]
```diff
@@ -9,6 +9,10 @@ ref:
     requests:
       cpu: 10m
       memory: 100Mi
-  credentials:
+  env:
+  - name: ENABLE_DES_DEFAULT_MACHINE
+    default: "false"
+    documentation: |-
+      Adds provisioned disk ecncryption set section under DefaultMachinePlatform in the install-config.yaml
```
Comment (gpei) [inline ci-operator/step-registry/ipi/conf/azure/provisioned-des/ipi-conf-azure-provisioned-des-ref.yaml:16]: a typo in the documentation 
"disk ecncryption set" -> "disk encryption set"

Comment (gpei) [reply ci-operator/step-registry/ipi/conf/azure/provisioned-des/ipi-conf-azure-provisioned-des-ref.yaml:16]: maybe we can update the documentation a bit more clearly like:
```
Adds provisioned disk encryption set section under DefaultMachinePlatform in the install-config.yaml:
* "false" (default) - Adds provisioned disk encryption set section under compute/controlPlane 
* "true" - Adds provisioned disk encryption set section under DefaultMachinePlatform, which also encrypts the default storage class from 4.13
```

Comment (jinyunma) [reply ci-operator/step-registry/ipi/conf/azure/provisioned-des/ipi-conf-azure-provisioned-des-ref.yaml]: updated

**Comment by jinyunma** [comment]
cc @liangxia to review

**Pull request merged**